class-voicewp.php

```diff
@@ -88,38 +118,15 @@ public function rest_pre_dispatch( $null, $that, $request ) {
 	 */
 	public function voicewp_skill_request( WP_REST_Request $request ) {
 
-		$this->voicewp_maybe_display_notice();
-
-		$body = $request->get_body();
-
 		$id = absint( $request->get_param( 'id' ) );
+		// get config based on url
+		$app_id = get_post_meta( $id, 'voicewp_skill_app_id', true );
 
-		if ( ! empty( $body ) ) {
-			try {
-				// get config based on url
-				//This ID will be made optional, first do a check for if standalone or not
-				$app_id = get_post_meta( $id, 'voicewp_skill_app_id', true );
-				$certificate = new \Alexa\Request\Certificate( $request->get_header( 'signaturecertchainurl' ), $request->get_header( 'signature' ), $app_id );
-				$alexa = new \Alexa\Request\Request( $body, $app_id );
-				$alexa->set_certificate_dependency( $certificate );
-
-				// Parse and validate the request.
-				$alexa_request = $alexa->from_data();
-
-			} catch ( InvalidArgumentException $e ) {
-				return $this->fail_response( $e );
-			}
-
-			$response_object = new \Alexa\Response\Response;
-			$event = new AlexaEvent( $alexa_request, $response_object );
+		list( $request, $response ) = $this->voicewp_get_request_response_objects( $request, $app_id );
```

Comment:
If the body is empty then this will not return an array of two values but instead null.  Maybe check the response of the function to make sure that it is not empty and is an array.